### ISSUE TITLE
Add tokens for collection rule actions

### ISSUE THREAD

**Issue by wiktork** [dotnet/dotnet-monitor#893]
This token will allow referencing previous actions output from configuration.
- This should also establish a dependency between actions once the async work is finished.

Reopening this issue to track other token work
- https://github.com/dotnet/dotnet-monitor/pull/1870 Adds Process.InstanceId

Also want to add:
Process.Pid
Process.Name

Environment.Key.

We should also consider tokenization for Egress provider configuration, such as the BlobPrefix.

**Comment by jander-msft** [comment]
Substitution tokens for process Id and process name were added in #3759 for 8.0 Preview 2. There has not been a customer ask for environment tokens. If other tokens are requested, a new issue should be opened to describe what should be added.
